### PR TITLE
Dod validation tests

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by ChrisBr** [comment]
LGTM :+1:

File: src/api/spec/models/download_repository_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
+
+RSpec.describe DownloadRepository do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:url) }
+    it { is_expected.to validate_presence_of(:arch) }
+    it { is_expected.to validate_presence_of(:repotype) }
+    it { is_expected.to validate_presence_of(:repository_id) }
+    it { expect(create(:download_repository)).to validate_uniqueness_of(:arch).scoped_to(:repository_id) }
+    it { is_expected.to validate_inclusion_of(:repotype).in_array(["rpmmd", "susetags", "deb", "arch", "mdk"]) }
```
Comment (mdeniz) [inline src/api/spec/models/download_repository_spec.rb]: Can you use DownloadRepository::REPOTYPES instead of the literal array?

Comment (bgeuken) [inline src/api/spec/models/download_repository_spec.rb]: That would defeat the purpose of this test

Comment (mdeniz) [inline src/api/spec/models/download_repository_spec.rb]: But in the examples I have seen the whitelisted values is 1:1 with the validation itself... so, they have to be the same..
Or I misunderstood why do we need to have hardcoded valuer here....

**Pull request merged**